store-api-server/src/payment/service/payment.service.ts

```diff
@@ -972,21 +975,34 @@ WHERE provider = 'simplex'
 
       events.push(event);
       for (const event1 of events) {
+        await this.createSimplexEventToDB(event1, paymentId);
         let deleteResponse = await deleteSimplexEvents(event1.event_id);
+
         if (deleteResponse?.status == 'OK') {
-          Logger.log(
-            `simplex payment DELETE event succeeded. eventId=${event1.event_id} paymentId=${paymentId}`,
-          );
+          Logger.log(`simplex payment DELETE event succeeded. eventId=${event1.event_id} paymentId=${paymentId}`);
         } else {
-          Logger.warn(
-            `simplex payment DELETE event failed. eventId=${event1.event_id} paymentId=${paymentId}`,
-          );
+          Logger.warn(`simplex payment DELETE event failed. eventId=${event1.event_id} paymentId=${paymentId}`);
         }
       }
     }
     Logger.log('getSimplexEvents FINISH successfully');
   }
 
+  private async createSimplexEventToDB(event1: any, paymentId: any) {
```

Comment:
Some code styling change requests:

- Would be slightly better to switch these parameters around. `paymentId` is the more pivotal parameter of the two here, `event1` is so to say "additional" info.
- `event1` is not a good name for a variable. Not very indicative and why the "1"? Why not something like: `eventJson` or `eventData`?
- `createSimplexEventToDb`, better to name this instead: `storeSimplexEvent`, more punctual and no less indicative.